sway/commands/bar.c

```diff
@@ -36,30 +34,27 @@ static struct cmd_handler bar_handlers[] = {
 
 // Must be in alphabetical order for bsearch
 static struct cmd_handler bar_config_handlers[] = {
-	{ "hidden_state", bar_cmd_hidden_state },
-	{ "mode", bar_cmd_mode }
+	{ "id", bar_cmd_id },
+	{ "swaybar_command", bar_cmd_swaybar_command },
 };
 
+// Determines whether the subcommand is valid in any bar handler struct
+static bool is_subcommand(char *name) {
+	return find_handler(name, bar_handlers, sizeof(bar_handlers)) ||
+		find_handler(name, bar_config_handlers, sizeof(bar_config_handlers));
+}
+
 struct cmd_results *cmd_bar(int argc, char **argv) {
 	struct cmd_results *error = NULL;
 	if ((error = checkarg(argc, "bar", EXPECTED_AT_LEAST, 1))) {
 		return error;
 	}
 
-	if (find_handler(argv[0], bar_config_handlers,
-				sizeof(bar_config_handlers))) {
-		if (config->reading) {
-			return config_subcommand(argv, argc, bar_config_handlers,
-					sizeof(bar_config_handlers));
-		}
-		return cmd_results_new(CMD_FAILURE, "bar",
-				"Can only be used in config file.");
-	}
-
+	bool spawn = false;
 	if (argc > 1) {
 		struct bar_config *bar = NULL;
-		if (!find_handler(argv[0], bar_handlers, sizeof(bar_handlers))
-				&& find_handler(argv[1], bar_handlers, sizeof(bar_handlers))) {
+		if (!is_subcommand(argv[0]) ||
```

Comment:
If a user wanted to define multiple bars in the config, this would force bar blocks again (expanded would work fine for one bar). Since there is no release allowing the `bar <bar-id> <subcommand>` syntax, this could be an option since there probably aren't many people who have the expanded notation in their config.

Reply:
Okay, it should be fine to keep the behaviour as it is.

However, is the first check `!is_subcommand(argv[0])` actually required? Can't think of a (valid) case when that would be true but the later check false.

Reply:
Done